vm/parser.ts

```diff
@@ -1070,6 +1070,19 @@ module J2ME {
     return cache[hash & 0xff];
   }
 
+  /**
+   * Generate an ID for a Class.  We use this to map Java objects to their
+   * ClassInfo objects, storing the ID for the Class in the first four bytes
+   * of the memory allocated for the Java object in the ASM heap.
+   *
+   */
+  export var generateClassId = (function() {
+    var nextId = 1;
+    return function() {
+      return nextId++;
+    }
+  })();
```

Comment:
Nit: I would actually move this into the *ClassInfo* class, making it a `private static` property of that class:

```
    /**
     * Generate an ID for a ClassInfo.  We use this to map Java objects to their
     * ClassInfo objects, storing the ID for the Class in the first four bytes
     * of the memory allocated for the Java object in the ASM heap.
     *
     */
    private static nextId: number = 1;
    private static generateClassId(): number {
      return ClassInfo.nextId++;
    }
```

Then, in the *ClassInfo* constructor:

```
      this.id = ClassInfo.generateClassId();
```

That way it's only exposed to the one class that calls it. And we use the encapsulation built into TypeScript to hide *nextId* rather than bespoke JavaScript encapsulation.

(You could even get rid of *generateClassId* and simply set *this.id* to `ClassInfo.nextId++`.)